src/Chapter2.hs

```diff
@@ -349,7 +355,9 @@ Implement a function that returns only the first half of a given list.
 "b"
 -}
 -- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
+firstHalf :: [a] -> [a]
+firstHalf l = take h l 
+  where h = div (length l) 2
```

Comment:
Great usage of `where` construction! Makes code easily readable 💯

Reply:
Thanks